src/ImageSharp/Formats/Webp/WebpChunkParsingUtils.cs

```diff
@@ -0,0 +1,351 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Buffers.Binary;
+using System.IO;
+using SixLabors.ImageSharp.Formats.Webp.BitReader;
+using SixLabors.ImageSharp.Formats.Webp.Lossy;
+using SixLabors.ImageSharp.Memory;
+using SixLabors.ImageSharp.Metadata;
+using SixLabors.ImageSharp.Metadata.Profiles.Exif;
+using SixLabors.ImageSharp.Metadata.Profiles.Xmp;
+
+namespace SixLabors.ImageSharp.Formats.Webp
+{
+    internal static class WebpChunkParsingUtils
+    {
+        /// <summary>
+        /// Reads the header of a lossy webp image.
+        /// </summary>
+        /// <returns>Information about this webp image.</returns>
+        public static WebpImageInfo ReadVp8Header(MemoryAllocator memoryAllocator, Stream stream, byte[] buffer, WebpFeatures features)
+        {
+            // VP8 data size (not including this 4 bytes).
+            stream.Read(buffer, 0, 4);
+            uint dataSize = BinaryPrimitives.ReadUInt32LittleEndian(buffer);
```

Comment:
We should probably check the length here.

Reply:
I have added length checks where appropriate